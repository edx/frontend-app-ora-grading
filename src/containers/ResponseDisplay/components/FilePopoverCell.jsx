import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from '@edx/frontend-platform/i18n';

import InfoPopover from 'components/InfoPopover';

import messages from './messages';

export const FilePopoverCell = ({ row: { original } }) => (
  <InfoPopover>
    <div className="help-popover-option">
      <strong><FormattedMessage {...messages.filePopoverNameTitle} /></strong>
      <br />
      {original.name}
    </div>
    <div className="help-popover-option">
      <strong><FormattedMessage {...messages.filePopoverDescriptionTitle} /></strong>
      <br />
      {original.description}
    </div>
  </InfoPopover>
);

FilePopoverCell.defaultProps = {
  row: {
    original: {},
  },
};

FilePopoverCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      downloadUrl: PropTypes.string,
    }),
  }),
};

export default FilePopoverCell;
