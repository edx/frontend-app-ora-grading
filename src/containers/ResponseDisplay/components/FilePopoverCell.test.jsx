import React from 'react';
import { shallow } from 'enzyme';

import FilePopoverCell from './FilePopoverCell';

jest.mock('components/InfoPopover', () => 'InfoPopover');

describe('FilePopoverCell', () => {
  describe('component', () => {
    const props = {
      row: {
        original: {
          name: 'some file name',
          description: 'long descriptive text...',
          downloadUrl: 'this-url-is.working',
        },
      },
    };
    let el;
    beforeEach(() => {
      el = shallow(<FilePopoverCell {...props} />);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });

    describe('behavior', () => {
      test('content', () => {
        const { original } = props.row;
        expect(el.text()).toContain(original.name);
        expect(el.text()).toContain(original.description);
      });
    });
  });
});
