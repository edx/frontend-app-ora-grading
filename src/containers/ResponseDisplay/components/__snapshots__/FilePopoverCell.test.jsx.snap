// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePopoverCell component snapshot 1`] = `
<InfoPopover>
  <div
    className="help-popover-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File Name"
        description="Popover title for file name"
        id="ora-grading.ResponseDisplay.FilePopoverCell.filePopoverNameTitle"
      />
    </strong>
    <br />
    some file name
  </div>
  <div
    className="help-popover-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File Description"
        description="Popover title for file description"
        id="ora-grading.ResponseDisplay.FilePopoverCell.filePopoverDescriptionTitle"
      />
    </strong>
    <br />
    long descriptive text...
  </div>
</InfoPopover>
`;
